libs/domain/search/src/services/resolvers/breadcrumbs.resolver.ts

```diff
@@ -0,0 +1,79 @@
+import { INJECTOR, Provider, inject } from '@spryker-oryx/di';
+import { Facet, FacetValue } from '@spryker-oryx/product';
+import { RouteType } from '@spryker-oryx/router';
+import {
+  Breadcrumb,
+  BreadcrumbsResolver,
+  BreadcrumbsResolvers,
+  LinkService,
+} from '@spryker-oryx/site';
+import {
+  Observable,
+  combineLatest,
+  map,
+  of,
+  switchMap,
+  throwError,
+} from 'rxjs';
+import { FacetListService } from '../facet-list.service';
+
+export type FlatFacet = { name: string; value: string };
+
+export class CategoryBreadcrumbsResolver implements BreadcrumbsResolver {
+  constructor(protected injector = inject(INJECTOR)) {}
+
+  protected getFlattenCategoriesBreadcrumbs(
+    categories: Facet
+  ): Observable<Breadcrumb[]> {
+    const getSelected = (facets?: FacetValue[]): FacetValue | undefined => {
+      return facets?.find(
+        ({ selected, children }) => selected || getSelected(children)
```

Comment:
Can we avoid recursive function?

Reply:
I asked **Denys** about existing limits of categories nesting and he answered that this was not limited. Due to that category facets have various nesting I don't see another way how to handle this without recursion. Any suggestion?